tools/server/webui/src/utils/tool_calling/js_repl_tool.ts

```diff
@@ -0,0 +1,72 @@
+import StorageUtils from '../storage';
+import { ToolCall, ToolCallOutput, ToolCallParameters } from '../types';
+import { AgentTool } from './agent_tool';
+
+export class JSReplAgentTool extends AgentTool {
+  private static readonly id = 'javascript_interpreter';
+  private fakeLogger: FakeConsoleLog;
+
+  constructor() {
+    super(
+      JSReplAgentTool.id,
+      () => StorageUtils.getConfig().jsInterpreterToolUse,
+      'Executes JavaScript code in the browser console. The code should be self-contained valid javascript. You can use console.log(variable) to print out intermediate values.',
+      {
+        type: 'object',
+        properties: {
+          code: {
+            type: 'string',
+            description: 'Valid JavaScript code to execute.',
+          },
+        },
+        required: ['code'],
+      } as ToolCallParameters
+    );
+    this.fakeLogger = new FakeConsoleLog();
+  }
+
+  _process(tc: ToolCall): ToolCallOutput {
+    const args = JSON.parse(tc.function.arguments);
+
+    // Redirect console.log which agent will use to
+    // the fake logger so that later we can get the content
+    const originalConsoleLog = console.log;
+    console.log = this.fakeLogger.log;
+
+    let result = '';
+    try {
+      // Evaluate the provided agent code
+      result = eval(args.code);
+      if (!result) {
+        result = '';
+      }
+    } catch (err) {
+      result = String(err);
+    }
```

Comment:
I think this can have many security risks, probably JS tool call is not the best choice. Btw, we already had python interpreter, why can't we just use that?

Reply:
I don't see a reason why both can't coexist (python and js). The reason I added js first is it was kinda easier.

I agree that in current setup this is not secure yet, but we can disable network requests and then it should be much better.

Reply:
Or at least a button "Allow this execution"

Reply:
I think most user won't want an `eval` hidden in the main code, even when it won't be called unless they explicitly enable it. At the very minimum, I think something equivalent iframe is required here.